OrbitGl/PresetsDataView.cpp

```diff
@@ -153,13 +199,45 @@ void PresetsDataView::DoFilter() {
 
 void PresetsDataView::OnDataChanged() {
   indices_.resize(presets_.size());
+  modules_.resize(presets_.size());
   for (size_t i = 0; i < presets_.size(); ++i) {
     indices_[i] = i;
+    std::vector<ModuleView> modules;
+    for (const auto& pair : presets_[i]->preset_info().path_to_module()) {
+      modules.push_back(
+          ModuleView(Path::GetFileName(pair.first), pair.second.function_hashes_size()));
+    }
+    modules_[i] = modules;
   }
 
   DataView::OnDataChanged();
 }
 
+bool PresetsDataView::GetDisplayColor(int row, int /*column*/, unsigned char& red,
+                                      unsigned char& green, unsigned char& blue) {
+  const std::shared_ptr<PresetFile> preset = GetPreset(row);
+  PresetLoadState load_state = GOrbitApp->GetPresetLoadState(preset);
+
+  switch (load_state) {
+    case kLoadable:
+      red = 200;
+      green = 240;
+      blue = 200;
+      return true;
+    case kPartiallyLoadable:
+      red = 240;
+      green = 240;
+      blue = 200;
+      return true;
+    case kNotLoadable:
+      red = 230;
+      green = 190;
+      blue = 190;
+      return true;
+  }
+  return false;
```

Comment:
this should be a free function just like the enum to_string function mentioned before.

Reply:
Move to the PresetLoadState struct, ptal